openvdb_points_clarisse/clarisse/module/Resource_OpenVDB_Points.cc

```diff
@@ -0,0 +1,489 @@
+}
+
+Geometry_OpenVDBPoints*
+create_openvdb_points_geometry(OfApp& application, ResourceData_OpenVDBPoints& data, const double fps, const bool overrideRadius, const double radius)
+{
+    PointDataGrid::Ptr grid = data.grid();
+
+    if (!grid)  return 0;
+    if (!grid->tree().cbeginLeaf())     return 0;
+
+    // check descriptor for velocity and radius (pscale)
+
+    PointDataTree::LeafCIter iter = grid->tree().cbeginLeaf();
+
+    assert(iter);
+
+    // retrieve motion blur length and direction
+
+    const double motionBlurLength = application.get_motion_blur_length();
+    const int motionBlurDirection = application.get_motion_blur_direction();
+    const AppBase::MotionBlurDirectionMode motionBlurMode =
+                        application.get_motion_blur_direction_mode_from_value(motionBlurDirection);
+
+    Geometry_OpenVDBPoints* geometry = Geometry_OpenVDBPoints::create(  grid, overrideRadius, radius,
+                                                                        fps, motionBlurLength, motionBlurMode);
+
+    if (geometry)
+    {
+        AppProgressBar* acc_progress_bar = application.create_progress_bar(CoreString("Computing VDB Point Acceleration Structures"));
+
+        geometry->computeAccelerationStructures();
+
+        acc_progress_bar->destroy();
+    }
+
+    return geometry;
+}
+
+GeometryPropertyCollection*
+create_openvdb_points_geometry_property(OfApp& application, ResourceData_OpenVDBPoints& data)
+{
+    GeometryPropertyCollection *property_array = new GeometryPropertyCollection;
+
+    CoreVector<GeometryProperty*> properties;
+
+    PointDataGrid::Ptr grid = data.grid();
+
+    if (!grid)  return property_array;
+
+    const PointDataTree& tree = grid->tree();
+
+    PointDataTree::LeafCIter iter = tree.cbeginLeaf();
+
+    if (!iter)  return property_array;
+
+    const AttributeSet::Descriptor& descriptor = iter->attributeSet().descriptor();
+
+    for (AttributeSet::Descriptor::ConstIterator it = descriptor.map().begin(),
+        end = descriptor.map().end(); it != end; ++it) {
+
+        const openvdb::NamePair& type = descriptor.type(it->second);
+
+        // only floating point property types are supported
+
+        if (type.first != "float" && type.first != "vec3s")   continue;
+
+        properties.add(new GeometryProperty_OpenVDBPoints(&data, it->first, type.first));
+    }
+
+    property_array->set(properties);
+
+    return property_array;
+}
+
+ParticleCloud*
+create_clarisse_particle_cloud(OfApp& application, ResourceData_OpenVDBPoints& data, const bool loadVelocities)
+{
+    const PointDataGrid::Ptr grid = data.grid();
+
+    if (!grid)                  return new ParticleCloud();
+
+    const PointDataTree& tree = grid->tree();
+
+    if (!tree.cbeginLeaf())     return new ParticleCloud();
+
+    const openvdb::Index64 size = pointCount(tree);
+
+    if (size == 0)              return new ParticleCloud();
+
+    CoreArray<GMathVec3d> array(size);
```

Comment:
Does clarisse store positions as doubles? I changed the position transform calculations to double precision to match the container, but init_positions() takes either a vector of GMathVec3d or GMathVec3f - does it just re-interpret based off the setter used? If so then storing as doubles is great, otherwise could just use a vector of GMathVec3f